**Association between Device Type and Customer Satisfaction**

**Background:**

Mizzare Corporation has collected data on customer satisfaction levels for two types of smart home devices: Smart Thermostats and Smart Lights. They want to determine if there's a significant association between the type of device purchased and the customer's satisfaction level.

**Data Provided:**

The data is summarized in a contingency table showing the counts of customers in each satisfaction level for both types of devices:

Satisfaction Smart Thermostat   Smart Light
                             
     

Total
Very Satisfied
50
70
120

Satisfied
80
100
180

Neutral
60
90
150

Unsatisfied
30
50
80

Very Unsatisfied
20
50
70

Total
240
360
600


**Objective:**

To use the Chi-Square test for independence to determine if there's a significant association between the type of smart home device purchased (Smart Thermostats vs. Smart Lights) and the customer satisfaction level.


Assignment Tasks:
1. State the Hypotheses:
2. Compute the Chi-Square Statistic:
3. Determine the Critical Value:
Using the significance level (alpha) of 0.05 and the degrees of freedom (which is the number of categories minus 1)
4. Make a Decision:
Compare the Chi-Square statistic with the critical value to decide whether to reject the null hypothesis.


1.State the hypothesis

Null Hypothesis: There is no significant association between the type of smart home device purchased (Smart Thermostats vs. Smart Lights) and the customer satisfaction level.

Alternative Hypothesis: There's a significant association between the type of smart home device purchased (Smart Thermostats vs. Smart Lights) and the customer satisfaction level.



In [1]:
# importing the necessary libraries
import numpy as np
from scipy.stats import chi2, chi2_contingency

In [2]:
# contingency table as a numpy array
observed_data = np.array([
    [50, 70],  # Very Satisfied
    [80, 100], # Satisfied
    [60, 90],  # Neutral
    [30, 50],  # Unsatisfied
    [20, 50]   # Very Unsatisfied
])

In [3]:
row_totals = observed_data.sum(axis=1) # calculates row total
col_totals = observed_data.sum(axis=0) # calculates column total
grand_total = observed_data.sum() # calculates grand total

In [4]:
# prints the row,column,grand totals
print("Row Totals:", row_totals)
print("Column Totals:", col_totals)
print("Grand Total:", grand_total)

Row Totals: [120 180 150  80  70]
Column Totals: [240 360]
Grand Total: 600


In [5]:
expected_data = np.outer(row_totals, col_totals) / grand_total # calculates the expected data

In [6]:
print(expected_data) # prints the expected data

[[ 48.  72.]
 [ 72. 108.]
 [ 60.  90.]
 [ 32.  48.]
 [ 28.  42.]]


2. Compute the Chi-Square Statistic:


In [7]:
chi_square_statistic = ((observed_data - expected_data) ** 2 / expected_data).sum() # calculates the chi-square statistic
chi_square_statistic

5.638227513227513

3. Determine the Critical Value:


In [8]:
alpha = 0.05 # sets the significance level
degrees_of_freedom = (observed_data.shape[0] - 1) * (observed_data.shape[1] - 1) # calculates dof
print(degrees_of_freedom)

4


In [9]:
critical_value = chi2.ppf(1 - alpha, degrees_of_freedom) # calculates critical value
critical_value

9.487729036781154

In [11]:
# we can directly use the package and perform the chisquare test
chi2_stat, p_val, dof, expected = chi2_contingency(observed_data)

print(f"Chi-Square Statistic: {chi2_stat}")
print(f"P-value: {p_val}")
print(f"Degrees of Freedom: {dof}")
print(f"Expected Frequencies: \n{expected}")

Chi-Square Statistic: 5.638227513227513
P-value: 0.22784371130697179
Degrees of Freedom: 4
Expected Frequencies: 
[[ 48.  72.]
 [ 72. 108.]
 [ 60.  90.]
 [ 32.  48.]
 [ 28.  42.]]


4. Make a decision

Since the calculated Chi-Square statistic (5.64) is less than the critical value (9.49), we fail to reject the null hypothesis.

Conclusion:

This means there is no significant association between the type of smart home device (Smart Thermostat or Smart Light) and customer satisfaction levels at the 5% significance level. Therefore, customer satisfaction appears to be independent of the type of device purchased.